src/compiler/crystal/tools/dependencies.cr

```diff
@@ -80,5 +112,67 @@ module Crystal
     private def match_patterns?(patterns, paths)
       patterns.any? { |pattern| paths.any? { |path| File.match?(pattern, path) } }
     end
+
+    class List < DependencyPrinter
+      private def print_file(filename, parent, filter, unseen)
+        @io.print path(filename)
+        if comment = edge_comment(filter, unseen)
+          @io.print " "
+          @io.print comment
+        end
+        @io.puts
+      end
+
+      private def print_indent
+        @io.print "  " * @stack.size unless @stack.empty?
+      end
+    end
+
+    class Dot < DependencyPrinter
+      def start_format
+        @io.puts "digraph G {"
+      end
+
+      def end_format
+        @io.puts "}"
+      end
+
+      private def print_file(filename, parent, filter, unseen)
+        return unless parent
+
+        @io.print "  "
+        @io.print path(parent)
+        @io.print " -> "
+        @io.print path(filename)
+        if comment = edge_comment(filter, unseen)
+          @io.print %( [label="#{comment}"])
+        end
+        @io.puts
+      end
+
+      private def path(filename)
+        super.to_s.inspect
+      end
+    end
+
+    class Mermaid < DependencyPrinter
+      def start_format
+        @io.puts "graph LR"
+      end
+
```

Comment:
A newline would be welcome :)

```suggestion
      def start_format
        @io.puts "graph LR"
      end

      def end_format
        @io.puts
      end

```

Reply:
Why? Every line already terminates with a newline character. So I believe this would only add an empty line at the end.

Reply:
Wouldn't this be a welcome visual separator between the printed entities?

Reply:
Between which entities? `end_format` runs exactly *once*, after the entire semantic analysis and printing the dependencies is complete.

Reply:
Oh, my bad then.